cart/src/main/java/com/yas/cart/service/CartItemV2Service.java

```diff
@@ -79,4 +114,30 @@ private CartItemV2 updateExistingCartItem(CartItemV2PostVm cartItemPostVm, CartI
         existingCartItem.setQuantity(existingCartItem.getQuantity() + cartItemPostVm.quantity());
         return cartItemRepository.save(existingCartItem);
     }
-}
+
+    private void validateCartItemDeleteVms(List<CartItemV2DeleteVm> cartItemDeleteVms) {
+        Map<Long, Integer> quantityByProductId = new HashMap<>();
+
+        for (CartItemV2DeleteVm cartItemDeleteVm : cartItemDeleteVms) {
+            Integer existingQuantity = quantityByProductId.get(cartItemDeleteVm.productId());
+
+            if (existingQuantity != null && !existingQuantity.equals(cartItemDeleteVm.quantity())) {
+                throw new BadRequestException(Constants.ErrorCode.DUPLICATED_CART_ITEMS_TO_DELETE);
+            }
+
+            quantityByProductId.put(cartItemDeleteVm.productId(), cartItemDeleteVm.quantity());
+        }
+    }
+
+    private Map<Long, CartItemV2> getCartItemsByProductIds(List<CartItemV2DeleteVm> cartItemDeleteVms) {
+        String currentUserId = AuthenticationUtils.extractUserId();
+        List<Long> productIds = cartItemDeleteVms
+            .stream()
+            .map(CartItemV2DeleteVm::productId)
+            .toList();
+        List<CartItemV2> cartItems = cartItemRepository.findWithLock(currentUserId, productIds);
+        return cartItems
+            .stream()
+            .collect(Collectors.toMap(CartItemV2::getProductId, Function.identity()));
```

Comment:
Can it throws an exception if car items have duplicate entities?

Reply:
Hi @minhtran-q, thank you for your feedback.

Since the table has a composite key on `customerId` and `productId`, this ensures that duplicates cannot occur for these columns. Therefore, the query will not return duplicate entities.

Reply:
okay, so let's skip it